test/cli.test.js

```diff
@@ -88,6 +88,42 @@ describe('CLI', () => {
     expect(() => JSON.parse(data)).not.toThrow();
   });
 
+  it('should be able to read template files with a UTF-8 BOM', async () => {
+    const utf8BOM = Buffer.from([0xEF, 0xBB, 0xBF]);
+
+    let files = [
+      'templates/alertRules.json',
+      'templates/autoscaleSettings.json',
+      'templates/servicePlan.json',
+      'templates/website.json'
+    ];
+
+    let mockFiles = new Map();
+
+    for (let file of files) {
+      let buf = Buffer.concat([utf8BOM, await readFile(file)]);
+      mockFiles.set(file, buf);
+    }
+
+    let rfn = mockFileRead(mockFiles);
+    let wfn = mockWriteFile();
+
+    let res = await executeCli('templates/*.json', 'merged-template.json');
+
+    rfn.mockRestore();
+
+    expect(res.exitCode).toBe(0);
+    expect(res.stderr).toBe('');
+
+    expect(wfn).toHaveBeenCalledTimes(1);
+    let [ outFilename, data ] = wfn.mock.calls[0];
+
+    wfn.mockRestore();
+
+    expect(outFilename).toBe('merged-template.json');
+    expect(() => JSON.parse(data)).not.toThrow();
```

Comment:
@nulltoken Yes. `JSON.parse()` seems to strictly adhere to the JSON grammar. If, besides whitespace, the parser finds something that is not a quoted string, a number, an array, an object, `true`, `false` or `null`, it will barf (I've tested it).

Reply:
@poveden To be honest, as I know (a little bit) your code style, I was actually searching in the test where you were actually checking that no BOM has found his way in the output. I guessed that this was caught by this last `JSON.parse()`.

So
 - The test does what it should
 - Maybe , a comment would make the intent of that call a little bit more explicit (nitpick)